pkg/api/store/cluster/cluster_store.go

```diff
@@ -534,6 +534,25 @@ func (r *Store) Update(apiContext *types.APIContext, schema *types.Schema, data
 	if err := r.validateUnavailableNodes(data, existingCluster, id); err != nil {
 		return nil, err
 	}
+
+	// Replace rancherKubernetesEngineConfig.cloudProvider.vsphereCloudProvider.virtualCenter value to properly update virtualCenter removal
+	if newVCenter, ok := values.GetValue(data, "rancherKubernetesEngineConfig", "cloudProvider", "vsphereCloudProvider", "virtualCenter"); ok && newVCenter != nil {
+		oldVCenter, ok := values.GetValue(existingCluster, "rancherKubernetesEngineConfig", "cloudProvider", "vsphereCloudProvider", "virtualCenter")
+		if ok && oldVCenter != nil && !reflect.DeepEqual(newVCenter, oldVCenter) {
+			newData := map[string]interface{}{}
+			for k,v := range data {
```

Comment:
nit/not-necessary to fix: should be comma here in `k, v`, gofmt should have done that.